config/samples/sample-app-agent.yml

```diff
@@ -0,0 +1,146 @@
+    openshift.io/generated-by: OpenShiftNewApp
+  creationTimestamp: "2023-05-04T14:49:52Z"
+  generation: 3
+  labels:
+    app: quarkus-test-agent
+    app.kubernetes.io/component: quarkus-test-agent
+    app.kubernetes.io/instance: quarkus-test-agent
+  name: quarkus-test-agent
+  namespace: cryostat-test
+  resourceVersion: "43683"
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      deployment: quarkus-test-agent
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        openshift.io/generated-by: OpenShiftNewApp
+      creationTimestamp: null
+      labels:
+        deployment: quarkus-test-agent
+    spec:
+      containers:
+      - env:
+        - name: CRYOSTAT_AGENT_APP_NAME
+          value: agent-test
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: CRYOSTAT_AGENT_BASEURI
+          value: https://cryostat-sample.$(NAMESPACE).svc:8181
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: status.podIP
+        - name: CRYOSTAT_AGENT_CALLBACK
+          value: http://$(POD_IP):9977
+        - name: CRYOSTAT_AGENT_AUTHORIZATION
+          value: Bearer abcd1234
+        - name: KEYSTORE_PASS
+          valueFrom:
+            secretKeyRef:
+              key: KEYSTORE_PASS
+              name: cryostat-sample-keystore
+        - name: JAVA_OPTS_APPEND
+          value: |-
+            -Djavax.net.ssl.trustStore=/var/run/secrets/myapp/truststore.p12
+            -Djavax.net.ssl.trustStorePassword=$(KEYSTORE_PASS)
+        image: quay.io/andrewazores/quarkus-test@sha256:96f0f8bf0878a0843df08f9fec8ee9cf9c0ea09ee8c462b2c28ae504085a8b1b
+        imagePullPolicy: IfNotPresent
+        name: quarkus-test-agent
+        ports:
+        - containerPort: 10010
+          protocol: TCP
+        - containerPort: 8080
+          protocol: TCP
+        - containerPort: 8443
+          protocol: TCP
+        - containerPort: 9097
+          protocol: TCP
+        resources: {}
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /var/run/secrets/myapp/truststore.p12
+          name: truststore
+          subPath: truststore.p12
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      schedulerName: default-scheduler
+      securityContext:
+        runAsNonRoot: true
+      terminationGracePeriodSeconds: 30
+      volumes:
+      - name: truststore
+        secret:
+          defaultMode: 420
+          secretName: cryostat-sample-tls
```

Comment:
Some places are using suffix `cryostat-sample-`. This is the name of the Cryostat CR, that could take other names? Maybe we should call out/warn about this kinda of usage?

Reply:
It's the same name as in the sample CR (`make create_cryostat_cr`). IMO it's not a big deal since these are both really developer conveniences. In particular this sample app thing is not intended for use by end users.